managed/services/agents/mongodb.go

```diff
@@ -232,26 +237,26 @@ func defaultCollectors(collectAll bool) map[string]collectorArgs {
 			enabled:     true,
 			enableParam: "--collector.replicasetstatus",
 		},
-		// disabled until we have better information on the resources usage impact
 		"collstats": {
 			enabled:     collectAll,
 			enableParam: "--collector.collstats",
 		},
-		// disabled until we have better information on the resources usage impact
 		"dbstats": {
 			enabled:     collectAll,
 			enableParam: "--collector.dbstats",
 		},
-		// disabled until we have better information on the resources usage impact
 		"indexstats": {
 			enabled:     collectAll,
 			enableParam: "--collector.indexstats",
 		},
-		// disabled until we have better information on the resources usage impact
 		"topmetrics": {
 			enabled:     collectAll,
 			enableParam: "--collector.topmetrics",
 		},
+		"shards": {
+			enabled:     collectAll,
+			enableParam: "--collector.shards",
+		},
```

Comment:
we need to add check for version 2.41.1 to support this flag

Reply:
Added check for 2.41.1